### PR TITLE
Storage: add 'google.cloud.storage.hmac_keys.HMACKeyMetadata' resource class.

### PR DESCRIPTION
**Pull request by tseaver**
@jkwlui Note that this PR targets the `storage-add-hmac-key-support-feature` branch, which will be merged to `master` once all the steps in #7851 are completed.

Toward #7851.

### CODE REVIEW

**Comment by tseaver** [comment]
Unrelated [system test flake](https://source.cloud.google.com/results/invocations/80a7a6cf-472f-405a-b768-7373a21d674f/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-python%2Fpresubmit%2Fstorage/log).  I reopened #7384.

File: storage/google/cloud/storage/hmac_key.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                "State may only be set to one of: {}".format(
+                    ", ".join(self._SETTABLE_STATES)
+                )
+            )
+
+        self._properties["state"] = value
+
+    @property
+    def time_created(self):
+        """Retrieve the timestamp at which the bucket was created.
```
Comment (jkwlui) [inline storage/google/cloud/storage/hmac_key.py]: nit: "Retrieve the timestamp at which the ~~bucket~~ **HMAC** key was created"

File: storage/google/cloud/storage/hmac_key.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                  ``None`` if the bucket's resource has not been loaded
+                  from the server.
+        """
+        value = self._properties.get("timeCreated")
+        if value is not None:
+            return _rfc3339_to_datetime(value)
+
+    @property
+    def updated(self):
+        """Retrieve the timestamp at which the bucket was created.
```
Comment (jkwlui) [inline storage/google/cloud/storage/hmac_key.py]: Same as above.

File: storage/google/cloud/storage/hmac_key.py (hunk 1 above)
Comment (tseaver) [reply storage/google/cloud/storage/hmac_key.py]: I will fix this on the merged feature branch.

File: storage/google/cloud/storage/hmac_key.py (hunk 2 above)
Comment (tseaver) [reply storage/google/cloud/storage/hmac_key.py]: I will fix this on the merged feature branch.

**Pull request merged**